include/ansi_x931_aes128.c

```diff
@@ -0,0 +1,91 @@
+#include "ansi_x931_aes128.h"
+#include "tiny-AES128-C/aes.h"
+
+#include <string.h>
+#include <stdlib.h>
+
+// TODO: CROSS PLATFORM?
+
+// Some helpers for the prng
+void cgc_gen_block(cgc_prng*);
+void cgc_xor(const uint8_t*, const uint8_t*, uint8_t*);
+
+/**
+ * Initializes a PRNG from a given seed
+ * @param  seed Initial state of the PRNG
+ * @return      Generated PRNG
+ */
+cgc_prng cgc_init_prng(const uint8_t *seed) {
+    cgc_prng prng;
+    memcpy(&prng.state, seed, sizeof(cgc_aes_state));
+    memset(prng.random_data, 0, BLOCK_SIZE);
```

Comment:
I'd prefer if you memset the whole structure at initialization and then populated state and random_idx.